### PR TITLE
internal/charmstore: fix PromulgatedURL so that it doesn't contain dev channel

### PR DESCRIPTION
**Pull request by rogpeppe**
This fixes a bug where the development channel was appearing even though
the charm was published.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1482/
Test PASSed.

File: internal/charmstore/search.go [hunk 1]
```diff
@@ -463,6 +463,7 @@ func (s *Store) syncSearch() error {
 			return errgo.Notef(err, "cannot index %s", rurl)
 		}
 	}
+	logger.Infof("finished sync search")
```
Comment (fabricematrat) [inline internal/charmstore/search.go:466]: thanks for that one !
We were also thinking to not stop on error (may be another PR?)

File: internal/v5/api.go [hunk 2]
```diff
@@ -781,6 +781,7 @@ func (h *ReqHandler) metaRevisionInfo(id *router.ResolvedURL, path string, flags
 	}
 	var response params.RevisionInfoResponse
 	for _, doc := range docs {
+		// TODO use proper promulgated URL.
```
Comment (fabricematrat) [inline internal/v5/api.go:784]: What is a proper promulgated url ?

**Comment by fabricematrat** [comment]
:+1:  with some questions just for me.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1483/
Test PASSed.

File: internal/v5/api.go (hunk 2 above)
Comment (rogpeppe) [inline internal/v5/api.go:784]: The promulgated URL that might have the dev channel in it.

**Comment by mhilton** [comment]
:+1:

**Comment by fabricematrat** [comment]
:+1:

File: internal/router/router.go [hunk 3]
```diff
@@ -192,6 +192,17 @@ func (id *ResolvedURL) PreferredURL() *charm.URL {
 	return u
 }
 
+// DocPromulgatedURL the promulgated URL that
```
Comment (frankban) [inline internal/router/router.go:195]: DocPromulgatedURL returns...

**Comment by rogpeppe** [comment]
:shipit:

File: internal/router/router.go [hunk 4]
```diff
@@ -192,6 +192,17 @@ func (id *ResolvedURL) PreferredURL() *charm.URL {
...
+// should be stored in the mongodoc.Entity.
+// It does not have the channel set.
+func (id *ResolvedURL) DocPromulgatedURL() *charm.URL {
```
Comment (frankban) [inline internal/router/router.go:198]: Not sure about this name, but I don;t have great alternatives. DocPromulgatedURL is like "use this only when you are working with mongo docs". Something like PromulgatedURLWithoutChannel would make the method reusable in other contexts. Take it or leave it.

**Comment by frankban** [comment]
:+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1484/
Test PASSed.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge
